### PR TITLE
add dpdispatcher

### PR DESCRIPTION
**Pull request by njzjz**
Add `dpdispatcher`, the new dependency on the existing `dpgen` package

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by njzjz** [comment]
@conda-forge/help-python ready for review

File: recipes/dpdispatcher/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - tqdm
+
+  run:
+    - python >=3.6
+    - dargs
+    - paramiko
+    - requests
+    - tqdm
+
+test:
```
Comment (synapticarbors) [inline recipes/dpdispatcher/meta.yaml:36]: Can you test the entry point with `commands`:
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-commands

File: recipes/dpdispatcher/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+  entry_points:
+    - dpdisp = dpdispatcher.dpdisp:main
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+    - setuptools_scm
+    - dargs
```
Comment (synapticarbors) [inline recipes/dpdispatcher/meta.yaml:24]: It looks like the `host` section probably only needs `python`, `pip` and `setuptools_scm`. I think the rest are just listed under `install_requires`, not `setup_requires`.

Comment (njzjz) [reply recipes/dpdispatcher/meta.yaml]: Done.

File: recipes/dpdispatcher/meta.yaml (hunk 1 above)
Comment (njzjz) [reply recipes/dpdispatcher/meta.yaml:32]: Done.

**Pull request merged**